libsnark/gadgetlib1/gadgets/basic_gadgets.tcc

```diff
@@ -261,29 +261,16 @@ void dual_variable_gadget<FieldT>::generate_r1cs_witness_from_bits()
 template<typename FieldT>
 void disjunction_gadget<FieldT>::generate_r1cs_constraints()
 {
-    /* inv * sum = output */
-    linear_combination<FieldT> a1, b1, c1;
-    a1.add_term(inv);
-    for (size_t i = 0; i < inputs.size(); ++i) {
-        b1.add_term(inputs[i]);
-    }
-    c1.add_term(output);
+    linear_combination<FieldT> sum = pb_sum(inputs);
 
+    // inv * sum = output
     this->pb.add_r1cs_constraint(
-        r1cs_constraint<FieldT>(a1, b1, c1),
+        r1cs_constraint<FieldT>({inv}, sum, {output}),
```

Comment:
I think this branch needs rebasing. I remember commenting about these curly brackets in another PR.

Reply:
Rebased onto `origin/develop`.  Thanks.  Marking for re-review.